### PR TITLE
Fix error spam from buttons

### PR DESCRIPTION
**Pull request by nickvergessen**

### CODE REVIEW

File: src/components/Button/Button.vue [hunk 1]
```diff
@@ -361,7 +361,7 @@ export default {
 
 	methods: {
 		getText() {
-			return this.$slots.default ? this.$slots.default[0].text.trim() : null
+			return this.$slots.default && this.$slots.default[0] && this.$slots.default[0].text ? this.$slots.default[0].text.trim() : null
```
Comment (nickvergessen) [inline src/components/Button/Button.vue:364]: Error in beforeUpdate hook: "TypeError: this.$slots.default[0].text is undefined"
found in
---> <Button>
TypeError: this.$slots.default[0].text is undefined
    getText Button.js:1
    beforeUpdate Button.js:1
    VueJS 6
        invokeWithErrorHandling
        callHook
        before
        flushSchedulerQueue
        nextTick
        flushCallbacks

File: src/components/Button/Button.vue [hunk 2]
```diff
@@ -250,13 +250,13 @@ export default {
...
-		 * Accepted values: primary, error, tertiary, warning, success. If left empty,
-		 * the default button style will be applyed.
+		 * Accepted values: primary, secondary, tertiary, error, warning, success. If left empty,
+		 * the default button style will be applied.
 		 */
 		type: {
 			type: String,
 			validator(value) {
-				return ['primary', 'tertiary', 'error', 'warning', 'success', ''].indexOf(value) !== -1
+				return ['primary', 'secondary', 'tertiary', 'error', 'warning', 'success', ''].indexOf(value) !== -1
```
Comment (nickvergessen) [inline src/components/Button/Button.vue:259]: Before:
```
[Vue warn]: Invalid prop: custom validator check failed for prop "type".

found in

---> <Button> at src/components/Button/Button.vue
       <LobbyStatus> at src/components/RightSidebar/LobbyStatus.vue
         <AppSidebar> at src/components/AppSidebar/AppSidebar.vue
           <RightSidebar> at src/components/RightSidebar/RightSidebar.vue
             <Content> at src/components/Content/Content.vue
               <App> at src/App.vue
                 <Root>
```

File: src/components/Button/Button.vue (hunk 1 above)
Comment (JuliaKirschenheuter) [inline src/components/Button/Button.vue:364]: Probably better this.$slots?.default[0]?.text?

Comment (nickvergessen) [reply src/components/Button/Button.vue]: Fixed

**Review by raimund-schluessler — APPROVED** [review]

**Review by marcoambrosini — APPROVED** [review]

File: src/components/Button/Button.vue (hunk 1 above)
Comment (nickvergessen) [reply src/components/Button/Button.vue]: Regression https://github.com/nextcloud/nextcloud-vue/pull/2547

Comment (korelstar) [reply src/components/Button/Button.vue]: @nickvergessen @JuliaKirschenheuter 
I think the correct expression would be:

`this.$slots?.default?.[0]?.text?.trim()`

Comment (nickvergessen) [reply src/components/Button/Button.vue]: That sadly didn't work 🤷‍♂️

Comment (korelstar) [reply src/components/Button/Button.vue]: Hmm. Okay, but did you noticed the `?.` directly behind of `default` ? That wasn't included in the commit. Though it is needed when `default` can be null/undefined.

**Pull request merged**